### PR TITLE
SSE-3228: frontend manage client section

### PR DESCRIPTION
**Pull request by Ryan-Andrews99**
### This PR Adds:
- Exposes new field on the frontend for the user to update, `back_channel_logout_uri`
- Exposes new field on the frontend for the user to update, `sector_identifier_uri`
- Adds the new `Manage Client` section as per the prototype
image

- New views to confirm removal of a specific `redirect_uri` and `post_logout_redirect_uri`
- New controllers to handle submission of these forms
- Adds a more flexible uriInput macro - existing Url input macro was not flexible enough
- Adds new UI tests for the new fields exposed on the frontend

### This PR Modifies:
- The view for the change  `redirect_uris` is updated to match the [design prototype](https://app.mural.co/t/governmentdigitalservice1334/m/governmentdigitalservice1334/1711110953589/1f081b6d5bf5426ae8e66707cff1ca31d5de4d88?sender=5d95b0be-f1a8-465f-ba08-b4655a159ec3)
Screenshot 2024-04-03 at 17 51 13

- The view for the change  `post_logout_redirect_uris` is updated to match the [design prototype](https://app.mural.co/t/governmentdigitalservice1334/m/governmentdigitalservice1334/1711110953589/1f081b6d5bf5426ae8e66707cff1ca31d5de4d88?sender=5d95b0be-f1a8-465f-ba08-b4655a159ec3)
Screenshot 2024-04-03 at 17 51 46

- Modifies the URI validation middleware to accept a URI limit, which is implemented in several places in this PR.
- Updates existing UI tests for  `redirect_uri` features and `post_logout_redirect_uri` to reflect new functionality
- Moves Public Key and User Attributes to separate section

### This PR Removes:
- A test from `post_logout_redirect_uris` feature which specified a user clicking on the 'default GOV.UK sign out page' link which I couldn't find existing. This test was failing, and I'm not sure why it was previously passing 🤔 

[SSE-3228]: https://govukverify.atlassian.net/browse/SSE-3228?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ui-automation-tests/support/steps/input-steps.ts [hunk 1]
```diff
@@ -14,9 +14,12 @@ const fields = {
     "service name": "serviceName",
     "new password": "newPassword",
     "current password": "currentPassword",
-    "redirect uris": "redirectUris",
-    "post logout redirect uris": "redirectUris",
-    "change your public key": "serviceUserPublicKey"
+    "redirect uris": "redirectUri",
```
Comment (Ryan-Andrews99) [inline ui-automation-tests/support/steps/input-steps.ts:17]: Do we still require both `redirect uris` and `redirect uri` for our tests?

File: ui-automation-tests/support/steps/input-steps.ts [hunk 2]
```diff
@@ -14,9 +14,12 @@ const fields = {
...
+    "redirect uri": "redirectUri",
```
Comment (Ryan-Andrews99) [inline ui-automation-tests/support/steps/input-steps.ts:18]: Do we still require both `redirect uris` and `redirect uri` for our tests?

File: ui-automation-tests/acceptance-features/clients/change-redirect-uris.feature [hunk 3]
```diff
@@ -9,36 +9,106 @@ Feature: Users can change their redirect uris
...
+      When they click on the link that points to "/confirm-redirect-uri-removal?uriToRemove=https%3A%2F%2Ftestredirecturi2.gov.uk"
+      Then they should be redirected to a page with the title "Are you sure you want to remove this redirect URI? - GOV.UK One Login"
+      And they select the "Yes" radio button
+      When they click the Confirm button
+      Then they should be redirected to a page with the title "Manage redirect URIs - GOV.UK One Login"
+      And they should not see the text "https://testredirecturi2.gov.uk"
+      And they should see the text "Redirect URIs updated"
+      When they click on the "Back" link
+      Then they should be redirected to a page with the title "Client details - GOV.UK One Login"
+      And they should not see the text "https://testredirecturi2.gov.uk"
```
Comment (Ryan-Andrews99) [inline ui-automation-tests/acceptance-features/clients/change-redirect-uris.feature:114]: This is a good test matcher 👍

File: express/src/lib/validators/uris-validator.ts [hunk 4]
```diff
@@ -1,23 +1,32 @@
 import {ValidationResult} from "../../types/validation-result";
 
-export default function validateUris(urls: string[]): ValidationResult {
+export default function validateUris(urls: string[], urlLimit?: number, isGovUrl?: boolean, allowEmptyValue?: boolean): ValidationResult {
```
Comment (Ryan-Andrews99) [inline express/src/lib/validators/uris-validator.ts:3]: I think, given we no longer use the multi URI validation, we should simplify this function a little bit and remove some of the multi URI validation stuff

File: ui-automation-tests/support/steps/input-steps.ts (hunk 2 above)
Comment (BorisP-GDS) [reply ui-automation-tests/support/steps/input-steps.ts:17]: fixed

File: express/src/lib/validators/uris-validator.ts (hunk 4 above)
Comment (BorisP-GDS) [reply express/src/lib/validators/uris-validator.ts]: fixed

**Review by MartinH-DI — APPROVED** [review]
This is all good

**Review by rafal-surowiec-co — APPROVED** [review]

**Pull request merged**